theano/tensor/elemwise.py

```diff
@@ -1045,11 +1062,96 @@ def _c_all(self, node, nodename, inames, onames, sub):
             alloc_fortran = '0'
 
         alloc = ""
-        # We loop over the "real" outputs, i.e., those that are not
-        # inplace (must be allocated) and we declare/allocate/check
-        # them
-        for output, oname, odtype in izip(
-                real_outputs, real_onames, real_odtypes):
+
+        # Array referencing all inputs
+        ilength = len(_inames)
+        inArray = """
+        PyArrayObject **inRef[%(ilength)s];
+        """ % locals()
+        for iname, j in izip(_inames, range(len(_inames))):
+            inArray += """
+            inRef[%(j)i]=&%(iname)s;
+            """ % locals()
+
+        # Array referencing all outputs
+        olength = len(_onames)
+        outArray = """
+        PyArrayObject **outRef[%(olength)s];
+        """ % locals()
+        for oname, j in izip(_onames, range(len(onames))):
+            outArray += """
+            outRef[%(j)i]=&%(oname)s;
+            """ % locals()
+
+        alloc += inArray + outArray
+
+        # Declaration and initialization of the hardcoded olv_index
+        # used by the code generated in elemwise_cgen
+        alloc += """
+        int olv_index;
+        olv_index = %(j)i;
+        """ % locals()
+        if 'params' in sub:
```

Comment:
CAReduce call the c_all function of Elemwise without properly generating the sub. That seemed to be quite a corner case and always called on node without a destroy map so I chose to fix it like that rather than search how to make CAReduce generate properly the sub with params in it.